spec/controllers/dashboards_controller_spec.rb

```diff
@@ -148,36 +148,31 @@ def load_table(klass, options)
       end
 
       context 'when successful' do
-        it 'adds a new version record' do
+        it 'sets the new production version' do
           SiteMapperHelper.silence do
-            expect { post(:push) }.to change(Version, :count).by(1)
+            expect { post(:push) }.to change(Version, :count).by(0)
+            expect(Version.current_production).not_to eq(nil)
           end
         end
 
-        it 'sets the new production version number to the preview number' do
-          SiteMapperHelper.silence do
-            post(:push)
-          end
-
-          expect(Version.current_production.number).to eq(Version.current_preview.number)
-        end
-
         it 'updates production data' do
           SiteMapperHelper.silence do
             post(:push)
           end
           expect(flash.notice).to eq('Production data updated')
         end
+      end
 
-        it 'when is not successful does not add a new version' do
-          allow(Version).to receive(:create).and_return(Version.new)
-          expect { post(:push) }.to change(Version, :count).by(0)
+      context 'when is not successful' do
+        before do
+          version = instance_double('Version')
+          allow(Version).to receive(:current_preview).and_return(version)
+          allow(version).to receive(:update).and_return(version)
+          allow(version).to receive(:persisted?).and_return(false)
         end
 
-        it 'when is not successful returns an error message' do
-          allow(Version).to receive(:create).and_return(Version.new)
-          post(:push)
-          expect(flash.alert).to eq('Production data not updated, remains at previous production version')
+        it 'does not add a new version' do
+          expect { post(:push) }.to change(Version, :count).by(0)
```

Comment:
Not sure what we're testing here.
`Version.count` doesn't change if it's successful _or_ fails.
How can it fail?  
`version.update(production: true)` should never fail, and if it does it will fail silently.
Looks like only if `Version.current_preview` is `blank?` does our outcome change.
I think we should just test that path.  
What do you think?

Reply:
I've pulled out the test. It looks like we are already covering that condition on `line 137`.